sampleapp/src/main/scala/de/up/hpi/informationsystems/sampleapp/dactors/Customer.scala

```diff
@@ -97,34 +97,33 @@ object Customer {
 
       case AddStoreVisit.Request(storeId: Int, time: ZonedDateTime, amount: Double, fixedDiscount: Double, varDiscount: Double) =>
         addStoreVisit(storeId, time, amount, fixedDiscount, varDiscount) match {
-          case Success(_) => sender() ! AddStoreVisit.Success(Seq.empty)
+          case Success(_) => sender() ! AddStoreVisit.Success(Relation(Seq.empty))
           case Failure(e) => sender() ! AddStoreVisit.Failure(e)
         }
 
       case Authenticate.Request(passwordHash) =>
         if (authenticate(passwordHash)) {
-          sender() ! Authenticate.Success(Seq.empty)
+          sender() ! Authenticate.Success(Relation(Seq.empty))
         } else {
           sender() ! Authenticate.Failure(AuthenticationFailedException("failed to authenticate using password"))
         }
     }
 
-    def getCustomerInfo: Try[Record] = {
+    def getCustomerInfo: Try[Relation] = {
       val rowCount = relations(CustomerInfo).records.get.size
       if (rowCount > 1) {
         throw InconsistentStateException(s"this relation was expected to contain at maximum 1 row, but contained $rowCount")
```

Comment:
You want to have the `Try` in line 112 (not in line 117), so it can catch the `Throwable`:
```scala
def .. = Try{
  [...]
  throw InconsistentStateException(...)
  [...]
  Relation(...)
}
```